Solutions/Nasuni/Hunting Queries/FileDeleteEvents.yaml

```diff
@@ -0,0 +1,59 @@
+id: 64a3477e-d06f-4491-86a5-6f99702e267f
+name: Nasuni File Delete Activity
+description: |
+  'This query looks for file delete audit events generated by a Nasuni Edge Appliance.'
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes:
+      - Syslog
+tactics:
+  - Impact
+relevantTechniques:
+  - T1485
+query: |
+  Syslog 
+  | where SyslogMessage matches regex "(nasuni.)([0-9A-Za-z]{8}-[0-9A-Za-z]{4}-[0-9A-Za-z]{4}-[0-9A-Za-z]{4}-[0-9A-Za-z]{1})"
+  | extend SyslogMessageJson = extract('{(.*?)}', 0, SyslogMessage)
+  | extend event_details = parse_json(SyslogMessageJson)
+  | extend path_parts = parse_path(tostring(event_details.path))
+  | where event_details.event_type == "AUDIT_UNLINK"
+  | project
+      TimeGenerated
+      ,HostName
+      ,event_type = event_details.event_type
+      ,sAMAccountName = trim(@"(?s)^.*\\\s*", tostring(event_details.username))
+      ,domainName = trim_end(@"[\\]\S*", tostring(event_details.username))
+      ,filename = path_parts.Filename
+      ,directorypath = path_parts.DirectoryPath
+      ,ipaddr = event_details.ipaddr
+      ,sid = event_details.sid
+      ,volume_guid = event_details.volume
+      ,access_point = event_details.resource
+      ,primary_group_name = trim(@"(?s)^.*\\\s*", tostring(event_details.groupname))
+      ,new_path = event_details.newpath
```

Comment:
for hunting queries only, we need to extend entities separately. Please check this for reference: https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Azure%20Activity/Hunting%20Queries/AnalyticsRulesAdministrativeOperations.yaml

Reply:
this one is pending. Entity mapping might not work as expected if not declared as expected. We should extend entities at the end of the query. E.g. shared below for two entity types. We need to do this for each entity type you have in your query

| extend IP_0_Address = ipaddr
| extend Account_0_Name = sAMAccountName

Reply:
Will be making this suggested change myself, as author requested to merge this solution ASAP.